include/hdf5/io/iobase.h

```diff
@@ -28,27 +28,13 @@
 
 #include <string>
 #include <vector>
+#include<boost/optional.hpp>
 
-#include <H5Cpp.h>
-
-// Define handy macros to detect whether we are above 1.8.11 and/or 1.10.0
-// These versions introduce some important backward-incompatible changes in the
-// C++ API that we need to be aware of if we want to support these versions
-#undef HDF5_NEWER_THAN_1_8_11
-#undef HDF5_NEWER_THAN_1_10_0
-#if HDF5_VERSION_MAJOR == 1 && \
-     (HDF5_VERSION_MINOR > 10 || \
-      (HDF5_VERSION_MINOR == 10 && HDF5_VERSION_PATCH >= 1))
-#define HDF5_NEWER_THAN_1_10_0
-#endif
-#if HDF5_VERSION_MAJOR == 1 && \
-     (HDF5_VERSION_MINOR > 8 || \
-      (HDF5_VERSION_MINOR == 8 && HDF5_VERSION_PATCH >= 12))
-#define HDF5_NEWER_THAN_1_8_11
-#endif
+#include "hdf5/data_set.h"
+#include "hdf5/data_space.h"
+#include "hdf5/file.h"
 
 namespace shark {
-
 namespace hdf5 {
```

Comment:
As mentioned in the main body, I've moved these classes to the `io` directory, but not to a `shark::hdf5::io` namespace - thoughts on adding that additional namespace?

Reply:
Not needed I think. The codebase is not huge, and having an inner `hdf5` namespace already contextualises all its contents.